towhee/runtime/data_queue.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2021 Zilliz. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Tuple
+import threading
+from enum import Enum, auto
+
+from collections import deque, namedtuple
+
+
+class DataQueue:
+
+    def __init__(self, schema_info, max_size=0):
+        self._max_size = max_size
+        self._schema = _Schema(schema_info)
+        self._data = []
+        for col_type in self._schema.col_types():
+            if col_type == ColumnType.QUEUE:
+                self._data.append(_QueueColumn())
+            else:
+                self._data.append(_NoQueueColumn())
+            
+        self._sealed = False
+        self._size = 0
+        self._lock = threading.Lock()
+        self._not_full = threading.Condition(self._lock)
```

Comment:
by using these conditions wont we require a seperate thread for every queue?

Reply:
yeah, each node of the graph will have a thread